File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int size=nums.size();

        // initialize two variable: 1. left pointer 2. right pointer
        int leftPtr = 0, rightPtr = size-1;

        // ensure to check in valid range
        while(leftPtr<=rightPtr)
        {
            int mid = leftPtr+(rightPtr-leftPtr)/2;

            // if target is found in the mid index then return the mid index
            if(nums[mid] == target)
            {
                return mid;
            }
            // if the element in mid index > target then update rightPtr
            else if(nums[mid] > target)
            {
                rightPtr = mid-1;
            }
            // if the element in mid index < target then update leftPtr
            else
            {
                leftPtr = mid+1;
            }
        }
        return -1;
    }
};

File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int start = 0;
        int end = nums.length - 1;
        int answer = bs(start, end, nums, target);
        return answer;
    }
    public static int bs(int start, int end, int[] nums, int target) {
        int mid = start + (end-start)/2;

        // BASE CONDITION
        if (start > end) {
            return -1;
        }

        if (nums[mid] > target) {
            end = mid-1;
            return bs(start, end, nums, target);
        } else if (nums[mid] < target) {
            start = mid+1;
            return bs(start, end, nums, target);
        }
        return mid;
    }
}